cmd_search.c

```diff
@@ -58,7 +58,9 @@ int cmd_search(context_t *context) {
       "--title         DEPRECATED. Same as --name.\n"
       "--all           Require all conditions match a window. Default is --any\n"
       "--any           Windows matching any condition will be reported\n"
-      "--sync          Wait until a search result is found.\n"
+      "--sync [T [A]]  Wait T * A seconds until a search result is found.\n"
+      "                T - sync waiting time till next attempt, default: 0.5\n"
```

Comment:
In this case --sync works with no arguments as well as with arguments. Arguments are not mandatory.

What kind of benefit is there to split those in two separate flags? Are there any different future flags intended for --sync?

As for timing argument(s)... At first when i wanted to implement timing, i thought too, to have it in one argument, just time in seconds, but when was looking on the code, that the check is done with simple loop not select() type of function and there was a TODO comment, that it would be nice to let it be tunable i thought that that is nice to have it this way, since somebody already needs it also. Another thing which i realized later was, when using already this in scripts was, that it was really convenient, because the --sync did not make xdotool sleep the whole time till rechecking, but it could recheck on the intervals specified in command line, thus in cases if window popped up earlier than total duration, script could continue execution at that point, thus in overall execution time would be smaller.

I don't know about other users and their use cases, but for mine it turned out to work really nice, because when scripting xdotool, i know exactly what the script is written for and characteristics for the executed application, thus can make longer waiting time till next attempt and not burn unnecessarily CPU cycles. OTOH nowadays CPU cycles are not so important and maybe not for xdotool, maybe users don't care if total script execution is 25 seconds or 30, it really is hard to tell.

As far as for --sync flag. With this patch:
1) if just --sync is specified (works as was before, no flags needed),
2) --sync 5, for example, would wait 5 seconds in total and the flag would work the same way you suggest and user does not have to care of details,
3) --sync 5 10 would be for more precise configured use cases.